src/core/cpu_patches.cpp

```diff
@@ -534,6 +550,166 @@ static void GenerateTcbAccess(const ZydisDecodedOperand* operands, Xbyak::CodeGe
 #endif
 }
 
+static void GenerateINSERTQ(const ZydisDecodedOperand* operands, Xbyak::CodeGenerator& c) {
+    // INSERTQ Instruction Reference
+    // Inserts bits from the lower 64 bits of the source operand into the lower 64 bits of the
+    // destination operand No other bits in the lower 64 bits of the destination are modified. The
+    // upper 64 bits of the destination are undefined.
+
+    // There's two forms of the instruction:
+    // INSERTQ xmm1, xmm2, imm8, imm8
+    // INSERTQ xmm1, xmm2
+
+    // For the immediate form:
+    // Insert field starting at bit 0 of xmm2 with the length
+    // specified by [5:0] of the first immediate byte. This
+    // field is inserted into xmm1 starting at the bit position
+    // specified by [5:0] of the second immediate byte.
+
+    // For the register form:
+    // Insert field starting at bit 0 of xmm2 with the length
+    // specified by xmm2[69:64]. This field is inserted into
+    // xmm1 starting at the bit position specified by
+    // xmm2[77:72].
+
+    // A value of zero in the field length is defined as a length of 64. If the length field is 0
+    // and the bit index is 0, bits 63:0 of the source operand are inserted. For any other value of
+    // the bit index, the results are undefined.
+
+    bool immediateForm = operands[2].type == ZYDIS_OPERAND_TYPE_IMMEDIATE &&
+                         operands[3].type == ZYDIS_OPERAND_TYPE_IMMEDIATE;
+
+    if (operands[0].type != ZYDIS_OPERAND_TYPE_REGISTER ||
+        operands[1].type != ZYDIS_OPERAND_TYPE_REGISTER) {
+        ASSERT_MSG("operands 0 and 1 must be registers.");
+    }
+
+    const auto dst = ZydisToXbyakRegisterOperand(operands[0]);
+    const auto src = ZydisToXbyakRegisterOperand(operands[1]);
+    Xbyak::Xmm xmm_dst = *reinterpret_cast<const Xbyak::Xmm*>(&dst);
+    Xbyak::Xmm xmm_src = *reinterpret_cast<const Xbyak::Xmm*>(&src);
```

Comment:
I assume this is guaranteed for this instruction but you may want to add an assert for isXmm just in case.